Sources/Classes/in-app-purchase/MiniAppIAPModule.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import StoreKit
+
+public class MiniAppIAPModule: NSObject, SKProductsRequestDelegate, SKPaymentTransactionObserver {
+
+    public typealias ProductResponseCompletionHandler = ((Result<MAProductResponse, MAProductResponseError>)) -> Void
+    var productResponseHandlerObj: ProductResponseCompletionHandler?
+    var paymentProductObj: SKProduct?
+    private var models = [SKProduct]()
+    var productId: String = ""
+
+    public func initiateProductPayment(productId: String, completionHandler: @escaping ProductResponseCompletionHandler) {
+        self.productResponseHandlerObj = completionHandler
+        self.productId = productId
+        let productRequest = SKProductsRequest(productIdentifiers: Set([self.productId]))
+        productRequest.delegate = self
+        productRequest.start()
+    }
+
+    func displayPaymentScreen(product: SKProduct) {
+        SKPaymentQueue.default().add(self)
+        let payment = SKPayment(product: product)
+        SKPaymentQueue.default().add(payment)
+        paymentProductObj = product
+    }
+
+    public func productsRequest(_ request: SKProductsRequest, didReceive response: SKProductsResponse) {
+        DispatchQueue.main.async {
+            self.models = response.products
+            if response.products.count == 0 {
+                self.productResponseHandlerObj?(.failure(.productNotFound))
+            } else {
+                self.displayPaymentScreen(product: response.products[0])
+            }
+        }
+    }
+
+    public func paymentQueue(_ queue: SKPaymentQueue, updatedTransactions transactions: [SKPaymentTransaction]) {
+        transactions.forEach({
+            switch $0.transactionState {
+            case .purchasing, .deferred, .restored:
+                break
+            case .purchased:
+                SKPaymentQueue.default().finishTransaction($0)
+                sendTranactionDetails(transaction: $0, status: .purchased)
+            case .failed:
+                SKPaymentQueue.default().finishTransaction($0)
+                sendTranactionDetails(transaction: $0, status: .failed)
+            @unknown default:
+                MiniAppLogger.d("MiniApp Transaction Update: Default \($0.payment.productIdentifier)")
```

Comment:
What is the caae where this would get triggered? Should it send an error response to the mini app?

Reply:
@corycaywood , we pretty much handle all the cases here from [TransactionState](https://developer.apple.com/documentation/storekit/skpaymenttransactionstate), however, Apple throws warning like this,

```
Switch covers known cases, but 'SKPaymentTransactionState' may have additional unknown values, possibly added in future versions
Handle unknown values using "@unknown default"
```

So added this default case with logs